### ISSUE TITLE
Add SMTPAdapter

### ISSUE THREAD

**Issue by paulcsmith** [thoughtbot/bamboo#60]
Possible libraries to use

* https://github.com/nurugger07/hermes
* https://github.com/antp/mailer

**Comment by aselder** [comment]
I'm really looking forward to this adapter, since we use AWS's SMTP servers. I may try and start hacking on this. Anybody want to help?

**Comment by paulcsmith** [comment]
@aselder That would be really great. I haven't needed SMTP so it would be awesome if you worked on it in a real world scenario. I think @BlakeWilliams started work on it, but I'm not sure how far he got

**Comment by paulcsmith** [comment]
@aselder How are things going with the SMTP adapter. If you need any help feel free to ping me :)

**Comment by lpil** [comment]
:+1: I would love to be able to use this lib with AWS's SMTP service.

**Comment by paulcsmith** [comment]
@anellis This is a trickier issue, but it would be *super* helpful to have in Bamboo. If you have the time, creating an SMTP adapter with https://github.com/antp/mailer or https://github.com/nurugger07/hermes would be super awesome.

I'd be happy to help with anything you need. Here are some docs for getting started: https://github.com/thoughtbot/bamboo/blob/master/lib/bamboo/adapter.ex

**Comment by keichan34** [comment]
I'd like to see this too -- I'm currently using `mailer` but this looks like it'd be easier to use in my case. My 2¢ would be to not use `mailer` as the SMTP backend but instead use `gen_smtp` (`mailer` uses `gen_smtp` behind the scenes anyways) or hermes (although I can't provide any advice on that since I haven't used it before).

**Comment by kdisneur** [comment]
I started a PullRequest, using `mailer`, any help/feedbacks would be very welcome: https://github.com/thoughtbot/bamboo/pull/135 :)

@keichan34 I looked at this issue yesterday and your comment was not present yet. It's a shame because you seem right, using `gen_smtp` would be a good idea:

1. it would allow people to use `mailer` with a different configuration, if needed by another part of the application
2. It would allow to add features more easily into the `bamboo` adapter
3. `mailer` use its own templating system, so we need call more internal stuff. So, I guess it makes the current implementation more brittle
